src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2719,15 +2921,16 @@ protected SocketAddress testableLocalSocketAddress() {
         return cnxn.sendThread.getClientCnxnSocket().getLocalSocketAddress();
     }
 
-    private static ClientCnxnSocket getClientCnxnSocket() throws IOException {
-        String clientCnxnSocketName = System
-                .getProperty(ZOOKEEPER_CLIENT_CNXN_SOCKET);
+    private ClientCnxnSocket getClientCnxnSocket() throws IOException {
```

Comment:
yes it is required as instance method getClientConfig() is being accessed
or shall i pass clientConfig in getClientCnxnSocket argument to let it remain static ?